pkg/api/vlabs/validate.go

```diff
@@ -649,9 +649,6 @@ func (a *Properties) Validate(isUpdate bool) error {
 		}
 
 		if a.OrchestratorProfile.OrchestratorType == Kubernetes {
-			if i == 0 {
-				continue
```

Comment:
This was causing a bug where we would not check anything for the first agentpool